lib/bolt/config.rb

```diff
@@ -82,6 +88,16 @@ def initialize(boltdir, config_data, overrides = {})
         @transports[key] = transport.default_options
       end
 
+      default_config_data = {}
+      DEFAULT_CONFIG_PATHS.each do |default_config_file|
+        begin
+          default_config_data.merge!(Bolt::Util.read_config_file(default_config_file, [], 'default config') || {})
+        rescue Bolt::FileError => e
+          @logger.debug(e.message)
+        end
+      end
+      config_data = default_config_data.merge(config_data)
```

Comment:
This merge means that you cannot set some ssh config at one level and some at another similiarly all plugins would have to be configured at the same level. Is this desirable or should we do a deeper merge. Especially for plugin hooks arbitrarily deep merging is wrong but maybe one level of merge?

Reply:
In general, I'm used to a deep merge of SSH settings. Thinking about the `~/.ssh/config` files.

Reply:
@adreyer & @trevor-vaughan: I also wondered about that, but it's still not clear what the optimal balance of behaviors (predictable + desirable - undesirable - future-keys) should be.  

Some possibilities:

|  Strategy | Effected  | Comments |
| --- | --- | --- |
| Shallow merge | All settings, top-level |  Simplest, safest, probably inconvenient-est |
| Shallow merge  | All settings, 1-deep  | Allows multiple transport opts and plugins |
| Deep merge | All settings | ⛔️ Not appropriate for `plugin_hooks` |
| Deep merge | All settings - undesirable | Shallow (or 1-d )for keys like `plugin_hooks` |
| Deep merge | Only desired settings | e.g., just transport settings  |
  
For the moment, I've left it as shallow top-level everything.

Reply:
@adreyer & @trevor-vaughan: I also wondered about that, but it's still not clear what the optimal balance of behaviors (predictable + desirable - undesirable - future-keys) should be.

Some possibilities:

|  Strategy | Effected  | Comments |
| --- | --- | --- |
| Shallow merge | All settings, top-level |  Simplest, safest, probably inconvenient-est |
| Shallow merge  | All settings, 1-deep  | Allows multiple transport opts and plugins |
| Deep merge | All settings | ⛔️ Not appropriate for `plugin_hooks` |
| Deep merge | All settings - undesirable | Shallow (or 1-d )for keys like `plugin_hooks` |
| Deep merge | Only desired settings | e.g., just transport settings  |
  
For the moment, I've left it as shallow top-level everything because it's easy and safe.